File: Data Structures & Algorithms/two-integer-sum/submission-14.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> diffs;

        for (int i = 0; i < nums.size(); i++) {
            int diff = target - nums[i];

            if (diffs.count(diff)) {
                vector<int> indexes = {diffs[diff], i};
                return indexes;
            }
            
            diffs.insert({nums[i], i});
            
        }

    

      return {-1};
    }
};

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-14.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        prefixSums = {0: 1}
        l = len(nums)
        s = 0; res=0

        for i in range(l):
            s += nums[i]
            diff = s - k
            res += prefixSums.get(diff, 0)
            prefixSums[s] = 1 + prefixSums.get(s, 0)

        return res
